src/System.Runtime.InteropServices.RuntimeInformation/src/RuntimeInformation.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+
+namespace System.Runtime.InteropServices
+{
+    public static partial class RuntimeInformation
+    {
+        private const string FrameworkName = ".NET Core";
+        private static Architecture? s_processArch = null;
+        private static readonly bool s_is64BitProcess = IntPtr.Size == 8;
+
+        public static string FrameworkDescription
+        {
+            get
+            {
+                return FrameworkName;
+            }
+        }
+
+        public static Architecture ProcessArchitecture
+        {
+            get
+            {
+                if (null == s_processArch)
+                {
+                    if (Architecture.Arm == OSArchitecture)
```

Comment:
> On Windows, this should call GetSystemInfo and switch based on the result (similar to OSArchitecture property).

This is calling the OSArchitecture which in turn calls ```GetNativeSystemInfo``` to get the OS Arch accurately. And instead of depending on win api, I am checking the value of IntPtr set using WIN32 constant at build time by CLR, since this is the process architecture, build time constant should work.